### ISSUE TITLE
Intro flow

### ISSUE THREAD

**Issue by peralmered** [Dhebug/Encounter#1]
So, the current intro flow, as I have it documented here is like an attract loop:

1. Splash/Jingle
2. Title screen
3. Scores
4. "How to play"
5. Typewriter
...and loop back.

This loop continues until the player hits a key to start playing, in which case...? The typewriter sequence plays again, possibly with a check if the player hasn't seen it before, during the attract loop?

This means that the music switches during the typewriter sequence, with drop in volume, and then either

- Back to splash/jingle music

OR

- Keep the typewriter music going.

### A third way
My idea is to keep the music doing the more upbeat energized version during the entire attract loop, so no change in music when the typewriter sequence comes in as part of the attract loop, and no typing sound during the attract loop.

However, once the player hits a key to start, the typewriter sequence starts, with typewriter music and typing sounds (ideally with an option to press ESC to skip).

**Comment by Dhebug** [comment]
Regarding the Splash/Jingle, it's not part of the attract loop, it's loaded as soon as possible, shows the "Severn Software" and "Defence Force" logo, and then proceed in loading the intro itself, so we need a short music for the splash, like a 20th Fox Century, THX, Dolby, Thalion intro, but it's only for the boot.

Regarding the attract loop, I'm not sure having the typewriter type without the typing during the attract loop, and then again when pressing play would feel good.

What about this flow instead:
- Game boots
- The "Severn Software" and "Defence Force" logos are shown with a short music jingle
- The intro attract mode starts:
  - Shows the title screen
  - Shows the high scores
  - Shows the "How to play"
  - Shows the parallax sequence with the PI office, but *without* the actual typewriter sequence
  - Loops back to the title screen
- When the player presses a key, then:
  - If the parallax sequence had not yet been shown, then show that...
  - If the typewriter sequence has not been played yet, then play it with the typing sounds effects
- Then the game starts
  - Play the game until the Game Over 
- Start the outro
  - Ask the player name if it's a high score
  - Show the credits
- Back to the intro attract mode 

So basically what that does:
- If the attract mode is aborted early, the player still see the entire type writer sequence, but they can abort it as well
- If they had a game over and decided to try again, then start the game directly without watching the sequence again.

**Comment by peralmered** [comment]
Crazy idea for the typewriter sequence: Could I give you a list of VBLs, and you could ...I guess the best word is "quantize" the typewriter strokes to those VBLs. That way, I could use the typewriter strokes as part of the rhythm of the music. The goal isn't to make the typewriter sound like a drum machine, more like the person typing is caught up in the music and can't help themselves from typing in time with the music.

**Comment by Dhebug** [comment]
That's probably doable, right now I do have some randomized amount of delay, but I could probably randomize over some specific steps.
